### ISSUE TITLE
Integration of SolusVM and Virtualizor [Feature Request]

### ISSUE THREAD

**Issue by agentmishra** [FOSSBilling/FOSSBilling#645]
SolusVM and VIrtualizor panel integrations would be awesome

**Comment by jaapmarcus** [comment]
https://github.com/FOSSBilling/SolusVM

**Comment by agentmishra** [comment]
it does not show in my downloaded archive, not sure its working properly
also virtualizor is not there

**Comment by jaapmarcus** [comment]
> it does not show in my downloaded archive, not sure its working properly
> also virtualizor is not there

It is treathed as an separate module you should download it you self. We want to keep the code base as small as possible and not include non needed modules

**Comment by agentmishra** [comment]
aahh, so many modules are needed and solusvm, you mean is a non needed module

**Comment by jaapmarcus** [comment]
Only one for SolusVM is needed and you download it by cloning the repo…

**Comment by agentmishra** [comment]
Virtualizor module needed
any help?

**Comment by jaapmarcus** [comment]
> Virtualizor module needed
> any help?

I think you need write it your self

**Comment by agentmishra** [comment]
> > Virtualizor module needed
> > any help?
> 
> I think you need write it your self

if you go through what i have written earlier, i thing i had made a [Feature Request]

by the way why only virtualizor module, a whole new billing automation could be coded to match my preferences by me

**Comment by John-S4** [comment]
> if you go through what i have written earlier, i thing i had made a [Feature Request]

Yes, and it's a valid feature request, but it's not something that is going to be a high priority right now for the core contributors. There is still a lot of work that is going on with the core of FOSSBilling, to make it as secure, stable, and extensible as possible, and updating and maintaining a few modules that are already out there but need a bit of modernisation. 

That doesn't mean that a virtualizor integration is never going to happen, or that we wouldn't be really happy to receive pull requests and help out if someone else wants to put one together. It just means that at the moment there are only a handful of people actively contributing to the project and they all have other priorities. 

The fact that the feature request is now here hopefully means that someone else with an interest in the same module will come along and get something moving :)
 
> by the way why only virtualizor module, a whole new billing automation could be coded to match my preferences by me

Well, yes, you could do that, but I think that we can all agree that it would be a LOT less work to build an integration with an existing and constantly improving existing solution (FOSSB) than to start from scratch. 

It would also be useful to more people to contribute here, or to find other people interested in the same integration and get them to help out. That's the real joy of open source, the fact that multiple contributors can ultimately end up creating something great.

**Comment by jaapmarcus** [comment]
> > > Virtualizor module needed
> > > any help?
> > 
> > 
> > I think you need write it your self
> 
> if you go through what i have written earlier, i thing i had made a [Feature Request]

Good job. The feature request has been added to a long list of other jobs according the importance we will schedule in for a feature release when we think it is useful. 
> 
> by the way why only virtualizor module, a whole new billing automation could be coded to match my preferences by me

Go ahead it will take a *** ton of time... 

If you expect us to develop tomorrow a plugin that works with Virtualizator forget it.  If it will be done it might take a few months or ever longer. It is how it works.. 

If you want get something added to the core that currently  is missing there are 3 options.
- Build it your self
- Hire somebody to build it for you 
- Or "wait" for it.  And accept it will take 6 years to be completed. 

As I don't own a Virtualizor server. To even get starting it will allready cost me at least 9 dollar + VPS server to even do test runs and see if it is working. 

As any other "module" / extension like a payment provider, domain provider, control panel connection or custom module are really unlikely to be added soon for the main reason there are to much of them to add them by use. 

This is why the community must pitch in and develop the plugins they need and share it on the internet for free this is how opensource is build and by demanding users to get thing working right now. In that case FOSSBilling is better of without you...
